src/http/middleware/set-auth-headers.js

```diff
@@ -1,10 +1,24 @@
-
 // Sets auth headers if necessary
 
-function addAuthHeaders ({ bearerToken, headers, overrideHeaders=false }) {
-  if (overrideHeaders || !bearerToken) return
+function addAuthHeaders ({ 
+  auth,
+  bearerToken, 
+  headers, 
+  overrideHeaders=false, 
+}) {
+  if (overrideHeaders) return
+  let authToken = bearerToken
```

Comment:
Eek, a`let`!! 🙅‍♂️ 😅 

Don't be afraid to have two return statements, even if it's a bit redundant:
```
const basicAuthToken = encodeBasicAuth(auth)
if (basicAuthToken) return { ... }
if (bearerToken) return { ... }
```

Reply:
Alternatively with one:
```
const basicAuthToken = encodeBasicAuth(auth)
if (!(basicAuthToken || bearerToken)) return
const authHeader = basicAuthToken ? ...
return { ... }
```